src/Engine/ProtoCore/FFI/CLRDLLModule.cs

```diff
@@ -1395,6 +1399,85 @@ public FFIMethodAttributes(MethodInfo method, Dictionary<MethodInfo, Attribute[]
             }
         }
 
+        public FFIMethodAttributes(MethodBase method, Dictionary<MethodInfo, Attribute[]> getterAttributes = null)
+        {
+            if (method == null)
+                throw new ArgumentNullException("method");
+
+            FFIClassAttributes baseAttributes = null;
+            Type type = method.DeclaringType;
+            if (!CLRModuleType.TryGetTypeAttributes(type, out baseAttributes))
+            {
+                baseAttributes = new FFIClassAttributes(type);
+                CLRModuleType.SetTypeAttributes(type, baseAttributes);
+            }
+
+            if (null != baseAttributes)
+            {
+                HiddenInLibrary = baseAttributes.HiddenInLibrary;
+            }
+
+            Attribute[] atts;
+            if (method is MethodInfo mInfo && getterAttributes.TryGetValue(mInfo, out atts))
```

Comment:
won't this throw if `getterAttributes is null?

Reply:
It will not be `null` with `MethodInfo` but I'll add the null-check just in case. Thanks.